src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -484,10 +491,41 @@ private bool TryClose()
 
                 await Input.CompleteAsync();
 
-                // Make sure application func is completed before completing writer.
-                if (_appCompleted != null)
+                var appCompleted = _appCompleted?.Task ?? Task.CompletedTask;
+                if (!appCompleted.IsCompletedSuccessfully)
```

Comment:
I think you just want IsCompleted here, you don't care if it was successful.
```suggestion
                if (!appCompleted.IsCompleted)
```

Reply:
This TCS will only be completed successfully so it won't matter.

The reason why I prefer IsCompletedSuccessfully anyway is if there is ever a chance that it doesn't complete successfully, the other branch will await the TCS and throw the error. That won't happen if IsCompleted is used.